### ISSUE TITLE
Bug: Model can not insert Entity

### ISSUE THREAD

**Issue by natanfelles** [codeigniter4/CodeIgniter4#3368]
**Describe the bug**

Model is not inserting entity values.

The following code:

```php
<?php namespace App\Controllers;

use App\Entities\User;
use App\Models\Users;

class Home extends BaseController
{
	public function index()
	{
		$user = new User([
			'name' => 'John Doe',
			'email' => 'jd@mail.com',
			'password' => 'secret',
		]);

		$users = new Users();
		$users->insert($user);
	}
}

```

throws the exception:

**CodeIgniter\Database\Exceptions\DataException**
_There is no data to insert._ 

**CodeIgniter 4 version**
4.0.4

**Affected module(s)**
Model/Entity

**Expected behavior, and steps to reproduce if appropriate**
Expect the data to be inserted.

**Context**
 - OS: Ubuntu
 - Web server: PHP
 - PHP version: 7.4.8

**Comment by natanfelles** [comment]
See also: https://forum.codeigniter.com/thread-77096.html

**Comment by michalsn** [comment]
I couldn't reproduce your error. In the linked forum thread you're saying something about custom Entity class. Could you please provide us your Model and Entity?

**Comment by natanfelles** [comment]
Hello @michalsn !

Here is the model:

```php
<?php namespace App\Models;

use CodeIgniter\Model;
use Faker\Generator;

class Users extends Model
{
    protected $table = 'users';
    protected $allowedFields = [
        'name',
        'email',
        'password',
    ];
    protected $useTimestamps = true;
    protected $returnType = User::class;

    public function fake(Generator &$faker)
    {
        return [
            'name' => $faker->name,
            'email' => $faker->email,
            'password' => password_hash('password', PASSWORD_DEFAULT),
        ];
    }
} 
```

And here is the entity, return type of the model:

```php
<?php namespace App\Models;

use CodeIgniter\Entity;

class User extends Entity
{
    protected $id;
    protected $name;
    protected $email;
    protected $password;
} 
```

And the controller, where the `fake` is called to test:

```php
<?php namespace App\Controllers;

use App\Models\Users as UsersModel;

class Home extends BaseController
{
    public function index()
    {
        helper('test');
        $user = fake(UsersModel::class);
        var_dump($user);
    }
} 
```

**Comment by lonnieezell** [comment]
Actually, the following code example is not how it should work for quite a while: 

```
<?php namespace App\Models;

use CodeIgniter\Entity;

class User extends Entity
{
    protected $id;
    protected $name;
    protected $email;
    protected $password;
} 
```

You should not specify them as class properties. All you need is: 

```<?php namespace App\Models;

use CodeIgniter\Entity;

class User extends Entity
{

} 
```

The rest will be handled by the class, as they are put in the `$attributes` array.

**Comment by yassinedoghri** [comment]
Hello, I've encountered a similar issue, I believe this is

**Comment by yassinedoghri** [comment]
Hello, I've encountered a similar issue. Basically, the [Handling Business Logic](https://codeigniter.com/user_guide/models/entities.html#id6) section in the docs doesn't work anymore in 4.0.4. If I understood well, it's a regression as it was working fine in previous versions. I've updated CodeIgniter to the latest develop changes after seeing that it was fixed in https://github.com/codeigniter4/CodeIgniter4/pull/3377 and it works fine now.
So, correct me if I'm wrong, but should there be a new stable version to "restore" the docs?

Anyways, thanks for the fix @michalsn!
